### ISSUE TITLE
WIN 10 account problem

### ISSUE THREAD

**Issue by riccardoprandini01** [MutonUfoAI/pgina#112]
Hi,
I have an Open Ldap from NethServer .
I have added several groups and users mixed them.
I have also an administrative r/w account and open r account
Under win 10 i have 
1) Added Pgina fork green and ok
2) Connected to LDAP via r/w account ok
3) Tested Authentication OK user is found
4) Tested Authorization OK user is found in specified group
5) Tested Gateway here I have some problem.

`2018-04-23 18:09:52,286 [3348|6|INFO ] pGina.Service.Impl: SessionChange:7 6
2018-04-23 18:09:52,301 [3348|53|INFO ] pGina.Service.Impl: SessionChange: 7 -> SessionLogoff
2018-04-23 18:09:52,301 [3348|53|INFO ] pGina.Service.Impl: Logoff in progress 7
2018-04-23 18:09:52,317 [3348|53|INFO ] pGina.Service.Impl: CREDUIhelper:(7)
2018-04-23 18:09:52,317 [3348|53|INFO ] pGina.Service.Impl: User:? in session:7 is unknown to pGina
2018-04-23 18:09:52,317 [3348|53|INFO ] pGina.Service.Impl: Logoff completed 7
2018-04-23 18:09:52,473 [3348|6|INFO ] pGina.Service.Impl: SessionChange:7 2
2018-04-23 18:09:52,489 [3348|36|INFO ] pGina.Service.Impl: SessionChange: 7 -> ConsoleDisconnect
2018-04-23 18:09:52,707 [3348|6|INFO ] pGina.Service.Impl: SessionChange:8 1
2018-04-23 18:09:52,738 [3348|35|INFO ] pGina.Service.Impl: SessionChange: 8 -> ConsoleConnect
2018-04-23 18:09:53,020 [3348|27|DEBUG] RemoteLog[NativeLib]: [Provider.cpp:95] Starting service state helper thread
2018-04-23 18:09:53,020 [3348|30|DEBUG] RemoteLog[NativeLib]: [CredentialProviderFilter.cpp:74] CredentialProviderFilter::Filter
2018-04-23 18:09:53,051 [3348|32|DEBUG] RemoteLog[NativeLib]: [Provider.cpp:129] Provider::SetUsageScenario(1, 0x00000000)
2018-04-23 18:09:53,067 [3348|31|DEBUG] RemoteLog[NativeLib]: [CredentialProviderFilter.cpp:74] CredentialProviderFilter::Filter
2018-04-23 18:09:53,067 [3348|3|DEBUG] RemoteLog[NativeLib]: [Provider.cpp:270] Provider::Advise(0000029518A119E0, 0000000000000009) - provider events callback reference added
2018-04-23 18:09:53,082 [3348|9|DEBUG] RemoteLog[NativeLib]: [Provider.cpp:439] SerializedUserNameAvailable: No serialized creds set
2018-04-23 18:09:53,082 [3348|10|DEBUG] RemoteLog[NativeLib]: [Provider.cpp:439] SerializedUserNameAvailable: No serialized creds set
2018-04-23 18:09:53,098 [3348|13|DEBUG] RemoteLog[NativeLib]: [Provider.cpp:456] SerializedPasswordAvailable: No serialized creds set
2018-04-23 18:09:53,098 [3348|14|DEBUG] RemoteLog[NativeLib]: [Provider.cpp:473] SerializedDomainNameAvailable: No serialized creds set
2018-04-23 18:10:10,192 [3348|20|DEBUG] RemoteLog[NativeLib]: [Credential.cpp:509] Credential::Connect: Processing login for fun
2018-04-23 18:10:10,207 [3348|18|DEBUG] PluginDriver:65695425-9e24-43eb-91ac-07f558d152f6: New PluginDriver created
2018-04-23 18:10:10,238 [3348|18|ERROR] Abstractions: GetDomainInfo() Error:1355 Il dominio specificato non esiste o è impossibile contattarlo
2018-04-23 18:10:10,238 [3348|18|INFO ] pGina.Service.Impl: domain:DESKTOP-HQRULL1
2018-04-23 18:10:10,254 [3348|18|INFO ] pGina.Service.Impl: domain:DESKTOP-HQRULL1
2018-04-23 18:10:10,270 [3348|18|ERROR] Abstractions: GetDomainInfo() Error:1355 Il dominio specificato non esiste o è impossibile contattarlo
2018-04-23 18:10:10,286 [3348|18|INFO ] pGina.Service.Impl: domain check:[False] [False] [False]
2018-04-23 18:10:10,286 [3348|18|DEBUG] pGina.Service.Impl: Processing LoginRequest for: fun in session: 8 reason: Login
2018-04-23 18:10:10,317 [3348|18|DEBUG] pGina.Service.Impl: Program running as user:SYSTEM
2018-04-23 18:10:10,317 [3348|18|DEBUG] pGina.Service.Impl: Program running as user:SERVIZIO LOCALE
2018-04-23 18:10:10,317 [3348|18|DEBUG] pGina.Service.Impl: Program running as user:SERVIZIO DI RETE
2018-04-23 18:10:10,332 [3348|18|DEBUG] pGina.Service.Impl: Program running as user:UMFD-0
2018-04-23 18:10:10,332 [3348|18|DEBUG] pGina.Service.Impl: Program running as user:DWM-8
2018-04-23 18:10:10,332 [3348|18|DEBUG] pGina.Service.Impl: Program running as user:UMFD-8
2018-04-23 18:10:10,348 [3348|18|DEBUG] PluginDriver:65695425-9e24-43eb-91ac-07f558d152f6: Begin login chain, 1 stateful plugin(s).
2018-04-23 18:10:10,348 [3348|18|DEBUG] LdapPlugin: BeginChain
2018-04-23 18:10:10,364 [3348|18|DEBUG] LdapServer: Initializing LdapServer host(s): [192.168.2.78], port: 636, useSSL = True, useTLS = False, verifyCert = False
2018-04-23 18:10:10,364 [3348|18|DEBUG] LdapServer: Timeout set to 10 seconds.
2018-04-23 18:10:10,379 [3348|18|DEBUG] PluginDriver:65695425-9e24-43eb-91ac-07f558d152f6: Performing login process
2018-04-23 18:10:10,379 [3348|18|DEBUG] PluginDriver:65695425-9e24-43eb-91ac-07f558d152f6: Authenticating user fun, 2 plugins available
2018-04-23 18:10:10,379 [3348|18|DEBUG] PluginDriver:65695425-9e24-43eb-91ac-07f558d152f6: Calling 12fa152d-a2e3-4c8d-9535-5dcd49dfcb6d
2018-04-23 18:10:10,395 [3348|18|DEBUG] LocalMachine: AuthenticateUser(65695425-9e24-43eb-91ac-07f558d152f6)
2018-04-23 18:10:10,395 [3348|18|DEBUG] LocalMachine: Found username: fun
2018-04-23 18:10:10,411 [3348|18|INFO ] LocalMachine: Authenticated user: fun
2018-04-23 18:10:10,411 [3348|18|DEBUG] LocalMachine: AuthenticateUser: Mirroring group membership from SAM
2018-04-23 18:10:10,441 [3348|18|DEBUG] PluginDriver:65695425-9e24-43eb-91ac-07f558d152f6: 12fa152d-a2e3-4c8d-9535-5dcd49dfcb6d Succeeded
2018-04-23 18:10:10,441 [3348|18|DEBUG] PluginDriver:65695425-9e24-43eb-91ac-07f558d152f6: Calling 0f52390b-c781-43ae-bd62-553c77fa4cf7
2018-04-23 18:10:10,441 [3348|18|DEBUG] LdapPlugin: AuthenticateUser(65695425-9e24-43eb-91ac-07f558d152f6)
2018-04-23 18:10:10,457 [3348|18|DEBUG] LdapPlugin: Received username: fun
2018-04-23 18:10:10,457 [3348|18|DEBUG] LdapPlugin: Attempting authentication for fun
2018-04-23 18:10:10,473 [3348|18|DEBUG] LdapServer: Attempting to bind with DN uid=fun,ou=people,dc=directory,dc=nh
2018-04-23 18:10:10,473 [3348|18|DEBUG] LdapServer: Attempting bind as uid=fun,ou=people,dc=directory,dc=nh
2018-04-23 18:10:10,488 [3348|18|DEBUG] LdapServer: Successful bind to 192.168.2.78 as uid=fun,ou=people,dc=directory,dc=nh
2018-04-23 18:10:10,488 [3348|18|DEBUG] LdapServer: LDAP DN uid=fun,ou=people,dc=directory,dc=nh successfully bound to server, return success
2018-04-23 18:10:10,504 [3348|18|INFO ] LdapServer: ntplist:192.168.2.78 192.168.2.78 
2018-04-23 18:10:10,504 [3348|18|FATAL] LdapServer: Unable to convert "shadowmax" to seconds Valore troppo grande o troppo piccolo per un Int32.
2018-04-23 18:10:10,519 [3348|18|DEBUG] LdapServer: convert attrib:[cn] to [Fullname] value:[Fun Fun]
2018-04-23 18:10:10,536 [3348|18|DEBUG] LdapServer: convert attrib:[mail] to [Email] value:[fun@mail-srl.com]
2018-04-23 18:10:10,536 [3348|18|DEBUG] PluginDriver:65695425-9e24-43eb-91ac-07f558d152f6: 0f52390b-c781-43ae-bd62-553c77fa4cf7 Succeeded
2018-04-23 18:10:10,552 [3348|18|INFO ] PluginDriver:65695425-9e24-43eb-91ac-07f558d152f6: Successfully authenticated fun
2018-04-23 18:10:10,552 [3348|18|DEBUG] PluginDriver:65695425-9e24-43eb-91ac-07f558d152f6: Authorizing user fun, 1 plugins available
2018-04-23 18:10:10,552 [3348|18|DEBUG] PluginDriver:65695425-9e24-43eb-91ac-07f558d152f6: Calling 0f52390b-c781-43ae-bd62-553c77fa4cf7
2018-04-23 18:10:10,566 [3348|18|DEBUG] LdapPlugin: LDAP Plugin Authorization
2018-04-23 18:10:10,566 [3348|18|DEBUG] LdapServer: Attempting bind as uid=admin,ou=people,dc=directory,dc=nh
2018-04-23 18:10:10,582 [3348|18|DEBUG] LdapServer: Successful bind to 192.168.2.78 as uid=admin,ou=people,dc=directory,dc=nh
2018-04-23 18:10:10,582 [3348|18|DEBUG] LdapPlugin: User fun is (&(memberUid=fun)) cn=pc_users,ou=Groups,dc=directory,dc=nh
2018-04-23 18:10:10,598 [3348|18|DEBUG] PluginDriver:65695425-9e24-43eb-91ac-07f558d152f6: 0f52390b-c781-43ae-bd62-553c77fa4cf7 Succeeded
2018-04-23 18:10:10,598 [3348|18|INFO ] PluginDriver:65695425-9e24-43eb-91ac-07f558d152f6: Successfully authorized fun
2018-04-23 18:10:10,613 [3348|18|DEBUG] PluginDriver:65695425-9e24-43eb-91ac-07f558d152f6: Processing gateways for user fun, 2 plugins available
2018-04-23 18:10:10,613 [3348|18|DEBUG] PluginDriver:65695425-9e24-43eb-91ac-07f558d152f6: Calling 12fa152d-a2e3-4c8d-9535-5dcd49dfcb6d
2018-04-23 18:10:10,613 [3348|18|DEBUG] LocalMachine: AuthenticatedUserGateway(65695425-9e24-43eb-91ac-07f558d152f6) for user: fun
2018-04-23 18:10:10,629 [3348|18|DEBUG] LocalAccount[fun]: SyncToLocalUser()
2018-04-23 18:10:10,644 [3348|18|DEBUG] LocalAccount[fun]: Checking for groups to remove.
2018-04-23 18:10:10,676 [3348|18|DEBUG] LocalAccount[fun]: Checking for groups to add
2018-04-23 18:10:10,676 [3348|18|DEBUG] LocalAccount[fun]: End SyncToLocalUser()
2018-04-23 18:10:10,691 [3348|18|DEBUG] PluginDriver:65695425-9e24-43eb-91ac-07f558d152f6: Calling 0f52390b-c781-43ae-bd62-553c77fa4cf7
2018-04-23 18:10:10,691 [3348|18|DEBUG] LdapPlugin: LDAP Plugin Gateway
2018-04-23 18:10:10,707 [3348|18|DEBUG] LdapServer: Attempting bind as uid=admin,ou=people,dc=directory,dc=nh
2018-04-23 18:10:10,707 [3348|18|DEBUG] LdapServer: Successful bind to 192.168.2.78 as uid=admin,ou=people,dc=directory,dc=nh
2018-04-23 18:10:10,722 [3348|18|FATAL] LdapServer: GetUserAttribValue Error:Filtro di ricerca non valido.
2018-04-23 18:10:10,722 [3348|18|DEBUG] LdapPlugin: User fun  is not 
2018-04-23 18:10:10,722 [3348|18|INFO ] LdapPlugin: Adding user fun to local group S-1-5-32-545, due to rule "Always add to "S-1-5-32-545""
2018-04-23 18:10:10,739 [3348|18|FATAL] LdapServer: GetUserAttribValue Error:Filtro di ricerca non valido.
2018-04-23 18:10:10,739 [3348|18|DEBUG] LdapPlugin: User fun  is not 
2018-04-23 18:10:10,739 [3348|18|INFO ] LdapPlugin: Adding user fun to local group S-1-5-32-544, due to rule "Always add to "S-1-5-32-544""
2018-04-23 18:10:10,754 [3348|18|INFO ] PluginDriver:65695425-9e24-43eb-91ac-07f558d152f6: Successfully processed gateways for fun
2018-04-23 18:10:10,754 [3348|18|DEBUG] PluginDriver:65695425-9e24-43eb-91ac-07f558d152f6: End login chain, 1 stateful plugin(s).
2018-04-23 18:10:10,770 [3348|18|DEBUG] LdapPlugin: EndChain
2018-04-23 18:10:10,770 [3348|18|DEBUG] LdapServer: Closing LDAP connection to 192.168.2.78.
2018-04-23 18:10:10,770 [3348|18|INFO ] pGina.Service.Impl: ses add LocalProfilePath:[C:\Users\fun.DESKTOP-HQRULL1]
2018-04-23 18:10:10,786 [3348|18|INFO ] pGina.Service.Impl: add user fun to sessioninfo:8 GUID:65695425-9e24-43eb-91ac-07f558d152f6 CREDUI:false
2018-04-23 18:10:10,786 [3348|18|INFO ] pGina.Service.Impl: ses username:fun description:pGina created credui:False isLoggedIN:False
2018-04-23 18:10:10,817 [3348|19|DEBUG] RemoteLog[NativeLib]: [Credential.cpp:799] Credential::ReportResult(0x00000000, 0x00000000)
2018-04-23 18:10:10,942 [3348|6|INFO ] pGina.Service.Impl: SessionChange:8 5
2018-04-23 18:10:11,004 [3348|35|INFO ] pGina.Service.Impl: SessionChange: 8 -> SessionLogon
2018-04-23 18:10:14,394 [3348|23|DEBUG] RemoteLog[NativeLib]: [Provider.cpp:281] Provider::UnAdvise() - provider events callback reference released
2018-04-23 18:10:14,410 [3348|21|DEBUG] RemoteLog[NativeLib]: [Provider.cpp:102] Stopping service state helper thread (if necessary)
`

**2018-04-23 18:10:10,722 [3348|18|FATAL] LdapServer: GetUserAttribValue Error:Filtro di ricerca non valido.**
It says search filter not valid.

I think i'm missing something have I to play with LocalMachine plugin?
My needs are login add to users or admins group. 
Another inspection of the user says that it is not in any group via windows management console.

In a second step if possible i'll set password as expired every 30 day via LDAP an let him change from remote.

**Comment by MutonUfoAI** [comment]
The error is very clear your defined filter is not valid.
Use an ldap client like "LDAP Admin" and verify that your filter is working.

**Comment by riccardoprandini01** [comment]
Thanks for the help, I'll try another filter but it is the same filter in Authorization and Gateway.

**Comment by riccardoprandini01** [comment]
Bad news i have made some try with ldap admin.

I have also forced for the moment in gateway the user to be added to S-1-5-32-544 and S-1-5-32-545

`2018-04-26 09:54:57,881 [2888|28|DEBUG] LdapPlugin: LDAP Plugin Gateway
2018-04-26 09:54:57,881 [2888|28|DEBUG] LdapServer: Attempting bind as uid=admin,ou=people,dc=directory,dc=nh
2018-04-26 09:54:57,896 [2888|28|DEBUG] LdapServer: Successful bind to 192.168.2.78 as uid=admin,ou=people,dc=directory,dc=nh
2018-04-26 09:54:57,896 [2888|28|FATAL] LdapServer: GetUserAttribValue Error:Filtro di ricerca non valido.
2018-04-26 09:54:57,912 [2888|28|DEBUG] LdapPlugin: User fun  is not 
2018-04-26 09:54:57,912 [2888|28|INFO ] LdapPlugin: Adding user fun to local group S-1-5-32-545, due to rule "Always add to "S-1-5-32-545""
2018-04-26 09:54:57,928 [2888|28|FATAL] LdapServer: GetUserAttribValue Error:Filtro di ricerca non valido.
2018-04-26 09:54:57,928 [2888|28|DEBUG] LdapPlugin: User fun  is not 
2018-04-26 09:54:57,928 [2888|28|INFO ] LdapPlugin: Adding user fun to local group S-1-5-32-544, due to rule "Always add to "S-1-5-32-544""
2018-04-26 09:54:57,942 [2888|28|INFO ] PluginDriver:5d25820c-1002-49ea-9173-36650d07d50b: Successfully processed gateways for fun
2018-04-26 09:54:57,942 [2888|28|DEBUG] PluginDriver:5d25820c-1002-49ea-9173-36650d07d50b: End login chain, 1 stateful plugin(s).
2018-04-26 09:54:57,942 [2888|28|DEBUG] LdapPlugin: EndChain
2018-04-26 09:54:57,958 [2888|28|DEBUG] LdapServer: Closing LDAP connection to 192.168.2.78.

`

The add to group is not working


I have 2 error

2018-04-26 09:54:57,896 [2888|28|FATAL] LdapServer: GetUserAttribValue Error:Filtro di ricerca non valido.

I have no search filter this time so why an error?

2018-04-26 09:54:57,912 [2888|28|DEBUG] LdapPlugin: User fun  is not 


The login credential ar the same and i have no problem in previous stage

2018-04-26 09:54:57,771 [2888|28|INFO ] PluginDriver:5d25820c-1002-49ea-9173-36650d07d50b: Successfully authorized fun
2018-04-26 09:54:57,787 [2888|28|DEBUG] PluginDriver:5d25820c-1002-49ea-9173-36650d07d50b: Processing gateways for user fun, 2 plugins available
2018-04-26 09:54:57,787 [2888|28|DEBUG] PluginDriver:5d25820c-1002-49ea-9173-36650d07d50b: Calling 12fa152d-a2e3-4c8d-9535-5dcd49dfcb6d
